pyhf/paramsets.py

```diff
@@ -32,3 +34,35 @@ def expected_data(self, pars):
             axis=0
         )
 
+def reduce_paramset_requirements(paramset_requirements):
+    reduced_paramset_requirements = {}
+
+    # nb: normsys and histosys have different op_codes so can't currently be shared
+    param_keys = ['constraint',
+                  'n_parameters',
+                  'op_code',
+                  'inits',
+                  'bounds',
+                  'auxdata',
+                  'factors']
+
+    for param_name in list(paramset_requirements.keys()):
+        params = paramset_requirements[param_name]
+
+        combined_param = {}
+        for param in params:
+            for k in param_keys:
+                combined_param.setdefault(k, set([])).add(param[k])
+
+        for k in param_keys:
+            v = combined_param[k]
+            if len(v) != 1:
+                raise exceptions.InvalidNameReuse("Multiple values for '{}' ({}) were found for {}. Use unique modifier names or use qualify_names=True when constructing the pdf.".format(k, list(v), param_name))
+            else:
+                v = list(v)[0]
+                if isinstance(v, tuple): v = list(v)
```

Comment:
what's going on here? why is this needed?

Reply:
bounds/inits/etc.. are sets() to add to a set() to check for uniqueness. Lists cannot be added to sets. But then since we need these to be mutable later in the code, we convert the sets to a list.

Reply:
ok I was confused by the many casts.. maybe rewrite to 
```
comb_attr = v.pop()
if isinstance(comb_attr, tuple): comb_attr = list(comb_attr)
combined_param[k]  = comb_attr
```